server/src/main/java/org/opensearch/search/sort/MinAndMax.java

```diff
@@ -93,4 +95,36 @@ public static Comparator<MinAndMax<?>> getComparator(SortOrder order) {
         }
         return Comparator.nullsLast(cmp);
     }
+
+    /**
+     * Compare given object with min
+     */
+    public int compareMin(Object object) {
+        return compare(getMin(), object);
+    }
+
+    /**
+     * Compare given object with max
+     */
+    public int compareMax(Object object) {
+        return compare(getMax(), object);
+    }
+
+    private int compare(T one, Object two) {
+        if (one instanceof Long) {
+            return Long.compare((Long) one, (Long) two);
```

Comment:
Nit: Do we need casting anymore with generics?
```suggestion
            return Long.compare(one, (Long) two);
```

Reply:
I think we need it still.... It not compiling if we remove this casting